### PR TITLE
RDPP-7218: OpenAPI guide

### PR DESCRIPTION
**Pull request by wickedest**
Thank you for your contribution to the API Builder documentation.

## Describe the changes

Enter a brief description of your changes to communicate to the reviewers what you changed and why.

## Checklist for contributors

Before submitting this PR, please make sure:

* [ ] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [ ] You have signed the [Axway CLA](https://cla.axway.com/)
* [ ] You have verified the technical accuracy of your change
* [ ] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [ ] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)
* [ ] You have verified that all status checks have passed

_Put an x in the boxes that apply. This is simply a reminder of what we are going to look for before merging your change._

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,3 +7,59 @@ This repository contains all files for building and deploying the netlify site.
...
+
+## Style guide
+
+### Text buttons
+
+When writing about text buttons, these shall be **Bold**, and written exactly as they appear in the UI.
+
+1. **Apply**
+1. **Close**
+1. **Discad changes**
```
Comment (alasdairhurst) [inline README.md:19]: typo

File: plugins/remark-lint-validate-internal-links/index.js [hunk 2]
```diff
@@ -186,6 +186,7 @@ function verifyAnchor(file, node, ref) {
 		if (availableAnchors.includes(anchor)) {
 			cache[ref] = warnings.valid;
 		} else {
+			debugger;
```
Comment (mitko-pelov) [inline plugins/remark-lint-validate-internal-links/index.js:189]: We need to clean these up.

File: content/en/docs/guide_openapi/managing_apidocs.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+* Removing parameters
+* Renaming parameters (this is effectively removing them)
+* Changing parameter names (even capitalization)
+* Changing the request
+* Changing the response content-type or schema
+* Removing responses
+
+## Deleting OpenAPI documents
+
+TODO
```
Comment (mitko-pelov) [inline content/en/docs/guide_openapi/managing_apidocs.md:60]: I'll take this.

File: content/en/docs/developer_guide/flows/flow_triggers/oas_flow_trigger.md [hunk 4]
```diff
@@ -5,8 +5,6 @@ weight: 10
 date: 2021-10-27
 ---
```
Comment (mitko-pelov) [inline content/en/docs/developer_guide/flows/flow_triggers/oas_flow_trigger.md:6]: The whole file needs more work. We have to discuss what we want to include here, as Jamie already did include a portion of what we would normally find here under https://deploy-preview-69--api-builder-docs.netlify.app/docs/guide_openapi/flows/

File: content/en/docs/guide_openapi/managing_apidocs.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+* Changing parameter names (even capitalization)
+* Changing the request
+* Changing the response content-type or schema
+* Removing responses
+
+## Deleting OpenAPI documents
+
+Deleting you OpenAPI documents is one of these actions that should be taken with caution. It is an irreversible action that will have a big impact on your service as you are effectively removing a chunk from it. It is important to note that deleting an OpenAPI document will delete all associated flows too. Any non related flows will be preserved.
+
+TODO: Insert gif here showcasing:
```
Comment (mitko-pelov) [inline content/en/docs/guide_openapi/managing_apidocs.md:62]: TODO: When we come up with the exact process of creating these gifs, add a gif here showcasing the details underneath.

I've attempted to do this with screenshots, but it adds a lot of additional text and space with the screenshots itself, so a simple gif will be best.

File: content/en/docs/developer_guide/flows/flow_triggers/oas_flow_trigger.md (hunk 4 above)
Comment (mitko-pelov) [reply content/en/docs/developer_guide/flows/flow_triggers/oas_flow_trigger.md:6]: Fixed.

File: content/en/docs/guide_openapi/managing_apidocs.md (hunk 3 above)
Comment (mitko-pelov) [reply content/en/docs/guide_openapi/managing_apidocs.md]: Fixed.

File: content/en/docs/guide_openapi/upgrading.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+npx @axway/api-builder-openapi-upgrade
+```
+
+After running, the command, you should see output similar to this:
+
+```text
+API Builder OpenAPI upgrade, version 1.0.0
+
+✔ Created: /home/jamie/piper/rdpp-swagger/apidocs/openapi/myapi.json
+✔ Skipped installation, please run "npm install" manually
```
Comment (alasdairhurst) [inline content/en/docs/guide_openapi/upgrading.md:40]: It won't say skipped installation when running npx @axway/api-builder-openapi-upgrade

File: content/en/docs/performance_metrics/_index.md [hunk 7]
```diff
@@ -23,16 +23,16 @@ The performance of a flow depends significantly on what the flow is doing. Conne
...
 | Concurrency | 10 |
-| Total time | 115.498 seconds |
-| Bytes transferred | 61,000,001 |
-| Time per request (mean) | 11.550 ms |
-| Time per request (mean, across all requests) | 1.155 ms |
-| Memory idle | 12 MB |
-| Memory max | 12 MB |
-| **Requests per second** | **865.81** |
+| Total time | 75.487 seconds |
+| Bytes transferred | 54,200,001 |
```
Comment (alasdairhurst) [inline content/en/docs/performance_metrics/_index.md:30]: Up above it's talking about GreetFlow as a baseline which doesn't exist anymore so it's not clear what implementation is being used. There's also a very different quantity of data so i guess it's a different API?

**Review by alasdairhurst — CHANGES_REQUESTED** [review]

File: content/en/docs/performance_metrics/_index.md (hunk 7 above)
Comment (wickedest) [reply content/en/docs/performance_metrics/_index.md:30]: It was the Greet API.  It does exist if you know where to find it.  We may need to "generically" describe the API, e.g. "simple GET API with one parameter..."

File: content/en/docs/guide_openapi/managing_apidocs.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+Currently, **OpenAPI** flow-trigger supports the following specifications:
+
+* [OpenAPI 2.0](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md)
+* [OpenAPI 3.0](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md)
+
+If you do not already have an OpenAPI specification to manage, you can read our guide on [how to write OpenAPI documents](/docs/guide_openapi/writing_apidocs).
+
+## Importing OpenAPI documents
+
+Currently, only one **OpenAPI** document is supported for **OpenAPI** flow-trigger. Once loaded, you can [update](#updating-openapi-documents) or [delete](#deleting-openapi-documents) it as required.
```
Comment (alasdairhurst) [inline content/en/docs/guide_openapi/managing_apidocs.md:19]: Can import multiple now. Need to figure out what to say, and probably have some mention of how multiple docs are managed. Somewhere in this section we also need to mention how the filename is important and influences where the docs are hosted, and link to the section in the quick start page (which also needs updating)

Comment (alasdairhurst) [reply content/en/docs/guide_openapi/managing_apidocs.md:19]: Need to check the rest of the docs for references to the previous single spec behavior.

**Review by alasdairhurst — APPROVED** [review]

**Pull request merged**